selene-vscode/src/util.ts

```diff
@@ -84,13 +82,16 @@ function getSeleneFilenamePattern(): RegExp {
     }
 }
 
-async function fileExists(filename: string): Promise<boolean> {
-    return fs.promises.stat(filename)
-        .then(() => true)
-        .catch((error) => error.code !== "ENOENT")
+async function fileExists(filename: vscode.Uri): Promise<boolean> {
+    try {
+        await vscode.workspace.fs.stat(filename)
+        return true
+    } catch (err) {
+        return err != vscode.FileSystemError.FileNotFound
+    }
 }
 
-export async function downloadSelene(directory: string) {
+export async function downloadSelene(directory: vscode.Uri) {

```

Comment:
Are you sure this should be using the vscode fs? It should only ever be downloading to an AppData folder or similar, seems like the standard fs stuff would be fine.

Reply:
The value originally passed to this was `ExtensionContext.globalStoragePath`, which was a string containing the fs path. This was deprecated in favour of `ExtensionContext.globalStorageUri`. I felt it would probably be wise to utilise `globalStorageUri` itself as much as possible (also allows us to use `vscode.Uri.joinPath`), and then finally use the normal filesystem path at the end when writing using `fsWriteFileAtomic` (see: line 103).

You may also want to consider removing `fsWriteFileAtomic` completely some time in the future with the vscode fs write file function - I didn't include changing that in this PR because I was unsure on how suitable it was for the task of writing a stream for a large binary file. Why would you want to do this? To quote https://code.visualstudio.com/api/references/vscode-api#workspace, the vscode fs APIs run outside of the editor process, compared to using the Node.js APIs (which run inside the extension context), so it may perform better [although downloading is still pretty quick for now!]

Reply:
The reason for using vscode stuff makes sense in the context of the global path being that type. The reaon for the atomic file writes is so that if I start to update selene (which can happen when you first open a Lua file), close VS Code, then re-open, it should use the older selene. Before atomic file writes they would be corrupted. All an "atomic file write" is in this system is downloading to a temp location then moving it over I think.

Reply:
Mhm, I understand the reasoning behind it - I'm not sure the VSCode APIs are suitable for what you want right now, so it should be left as-is 👍